src/test/java/online/book/store/service/ShoppingCartServiceTest.java

```diff
@@ -0,0 +1,248 @@
+package online.book.store.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Optional;
+import online.book.store.dto.cartitem.CartItemRequestDto;
+import online.book.store.dto.cartitem.UpdateCartItemRequestDto;
+import online.book.store.dto.shoppingcart.ShoppingCartDto;
+import online.book.store.exception.EntityNotFoundException;
+import online.book.store.mapper.CartItemMapper;
+import online.book.store.mapper.ShoppingCartMapper;
+import online.book.store.model.Book;
+import online.book.store.model.CartItem;
+import online.book.store.model.ShoppingCart;
+import online.book.store.model.User;
+import online.book.store.repository.book.BookRepository;
+import online.book.store.repository.cartitem.CartItemRepository;
+import online.book.store.repository.shoppingcart.ShoppingCartRepository;
+import online.book.store.service.impl.ShoppingCartServiceImpl;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class ShoppingCartServiceTest {
```

Comment:
Try to increase readability using private methods for expected content and using Given-When-Then construction